Python/sysmodule.c

```diff
@@ -96,6 +96,80 @@ PySys_SetObject(const char *name, PyObject *v)
         return PyDict_SetItemString(sd, name, v);
 }
 
+static PyObject *
+sys_breakpointhook(PyObject *self, PyObject *args, PyObject *keywords)
+{
+    char *envar = getenv("PYTHONBREAKPOINT");
+
+    if (envar == NULL || strlen(envar) == 0) {
+        envar = "pdb.set_trace";
+    }
+    else if (!strcmp(envar, "0")) {
+        /* The breakpoint is explicitly no-op'd. */
+        Py_RETURN_NONE;
+    }
+    char *last_dot = strrchr(envar, '.');
+    char *attrname = NULL;
+    PyObject *modulepath = NULL;
+
+    if (last_dot == NULL) {
+        /* The breakpoint is a built-in, e.g. PYTHONBREAKPOINT=int */
+        modulepath = PyUnicode_FromString("builtins");
+        attrname = envar;
+    }
+    else {
+        /* Split on the last dot; */
+        modulepath = PyUnicode_FromStringAndSize(envar, last_dot - envar);
+        attrname = last_dot + 1;
+    }
```

Comment:
*If* you want to handle non-ASCII characters correctly, it seems like env var should be decoded using:

```
#ifdef __APPLE__
            /* Use utf-8 on Mac OS X */
            unicode = PyUnicode_FromString(p);
#else
            unicode = PyUnicode_DecodeLocale(p, "surrogateescape");
#endif
```

according to Modules/main.c code to handle PYTHONWARNINGS. In fact, it's even worse, this file also uses the following code on Windows to get directly Unicode:

``
!Py_IgnoreEnvironmentFlag && (wp = _wgetenv(L"PYTHONWARNINGS"))
```

Maybe leave it unchanged. It's ok to not support non-ASCII Python module/function names on non-UTF-8 locale encodings (really weird case).

Reply:
Agreed!  I'll leave this one alone.